src/util/blockchain.ts

```diff
@@ -73,116 +73,149 @@ export const CHAIN_IDS: ChainIds = {
-    case 110: // testnet
-    case 111:
-      return Blockchain.Zilliqa;
-    case 14: // mainnet
-    case 88: // testnet
-      return Blockchain.Neo3;
-    case 19: // mainnet
-      return Blockchain.Arbitrum;
-    case 244: // mainnet
-      return Blockchain.Osmosis;
-    case 245: // mainnet
-      return Blockchain.Terra;
-    case 246: // mainnet
-      return Blockchain.CosmosHub;
-    case 247: // mainnet
-      return Blockchain.Juno;
-    case 248: // mainnet
-      return Blockchain.Evmos;
-    case 249: // mainnet
-      return Blockchain.Axelar;
-    case 313: // mainnet
-      return Blockchain.Stride;
-    case 314: // mainnnet
-      return Blockchain.Kujira;
-    case 315: // mainnnet
-      return Blockchain.Terra2;
-    case 316: // mainnnet
-      return Blockchain.Quicksilver;
-    case 317: // mainnnet
-      return Blockchain.Comdex;
-    case 318: // mainnnet
-      return Blockchain.StafiHub;
-    case 319: // mainnnet
-      return Blockchain.Persistence;
-    case 320: // mainnnet
-      return Blockchain.Stargaze;
-    case 42161: // mainnnet
-      return Blockchain.Arbitrum;
+export const blockchainForChainId = (chainId?: number, network = Network.MainNet): Blockchain | undefined => {
+  switch (network) {
+    case Network.MainNet:
+      switch (chainId) {
+        case 0:
+          return Blockchain.Native
+        case 1:
+          return Blockchain.Btc
+        case 4:
+          return Blockchain.Neo
+        case 6:
+          return Blockchain.BinanceSmartChain
+        case 14:
+          return Blockchain.Neo3
+        case 9:  /* FALLTHROUGH */
+        case 18: /* FALLTHROUGH */
+        case 111:
+          return Blockchain.Zilliqa
+        case 244:
+          return Blockchain.Osmosis
+        case 13: /* FALLTHROUGH */
+        case 245:
+          return Blockchain.Terra
+        case 246:
+          return Blockchain.CosmosHub
+        case 5: /* FALLTHROUGH */
+        case 247:
+          return Blockchain.Juno
+        case 7: /* FALLTHROUGH */
+        case 248:
+          return Blockchain.Evmos
+        case 8: /* FALLTHROUGH */
+        case 249:
+          return Blockchain.Axelar
+        case 313:
+          return Blockchain.Stride
+        case 314:
+          return Blockchain.Kujira
+        case 315:
+          return Blockchain.Terra2
+        case 316:
+          return Blockchain.Quicksilver
+        case 12: /* FALLTHROUGH */
+        case 317:
+          return Blockchain.Comdex
+        case 318:
+          return Blockchain.StafiHub
+        case 15: /* FALLTHROUGH */
+        case 319:
+          return Blockchain.Persistence
+        case 16: /* FALLTHROUGH */
+        case 320:
+          return Blockchain.Stargaze
+        case 19: /* FALLTHROUGH */
+        case 42161:
+          return Blockchain.Arbitrum
+        default:
+          return undefined
+      }
+    case Network.TestNet:
+      switch (chainId) {
+        case 1:
+          return Blockchain.Terra
```

Comment:
Hmm on Carbon chain, chain-id 1 is BTC too. I'm not quite sure if this is going to work out... Might wanna check with Steven how to deal with this
<img width="378" alt="Screenshot 2023-02-01 at 3 38 30 PM" src="https://user-images.githubusercontent.com/65583522/215980505-809d54db-669e-47d7-bad6-f923f27d9030.png">

Reply:
why is 1 = terra on testnet? looks like a mistake?

Reply:
@stevenkhong chainId 1 via polynetwork is BTC, via IBC is Terra, not sure which to choose. [testnet bridges api](https://test-api.carbon.network/carbon/coin/v1/bridges?pagination.limit=10000)